diff --git a/configure b/configure
index 732de59292..136347a0d2 100755
--- a/configure
+++ b/configure
@@ -1886,8 +1886,6 @@ EXTERNAL_LIBRARY_GPL_LIST="
     libdvdread
     librubberband
     libvidstab
-    libx264
-    libx265
     libxavs
     libxavs2
     libxvid
@@ -1900,6 +1898,8 @@ EXTERNAL_LIBRARY_NONFREE_LIST="
 "

 EXTERNAL_LIBRARY_VERSION3_LIST="
+    libx264
+    libx265
     gmp
     libaribb24
     liblensfun
