--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -56,7 +56,7 @@
 set(POWER_ALIASES powerpc64 powerpc64le ppc64 ppc64le)
 list(FIND POWER_ALIASES "${SYSPROC}" POWERMATCH)
 list(FIND RISCV64_ALIASES "${SYSPROC}" RISCV64MATCH)
-if(X86MATCH GREATER "-1")
+if(X86MATCH GREATER "-1" AND NOT EMSCRIPTEN)
     set(X86 1)
     add_definitions(-DX265_ARCH_X86=1)
     if(CMAKE_CXX_FLAGS STREQUAL "-m32")
@@ -292,7 +292,7 @@
         else()
             add_definitions(-march=native)
         endif()
-    elseif(X86 AND NOT X64)
+    elseif(X86 AND NOT X64 AND NOT EMSCRIPTEN)
         string(FIND "${CMAKE_CXX_FLAGS}" "-march" marchPos)
         if(marchPos LESS "0")
             add_definitions(-march=i686)
