--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -56,7 +56,7 @@
 set(POWER_ALIASES powerpc64 powerpc64le ppc64 ppc64le)
 list(FIND POWER_ALIASES "${SYSPROC}" POWERMATCH)
 list(FIND RISCV64_ALIASES "${SYSPROC}" RISCV64MATCH)
-if(X86MATCH GREATER "-1")
+if(X86MATCH GREATER "-1" AND NOT EMSCRIPTEN)
     set(X86 1)
     add_definitions(-DX265_ARCH_X86=1)
     if(CMAKE_CXX_FLAGS STREQUAL "-m32")
@@ -292,7 +292,7 @@
         else()
             add_definitions(-march=native)
         endif()
-    elseif(X86 AND NOT X64)
+    elseif(X86 AND NOT X64 AND NOT EMSCRIPTEN)
         string(FIND "${CMAKE_CXX_FLAGS}" "-march" marchPos)
         if(marchPos LESS "0")
             add_definitions(-march=i686)
--- a/source/common/cpu.cpp
+++ b/source/common/cpu.cpp
@@ -159,0 +160,3 @@
+#if defined(__APPLE__) && X86_64
+    (void)benableavx512;
+#endif
@@ -219,0 +223 @@
+#if !defined(__APPLE__) || !X86_64
@@ -230,0 +235 @@
+#endif
--- a/source/common/x86/asm-primitives.cpp
+++ b/source/common/x86/asm-primitives.cpp
@@ -2364,0 +2365 @@
+#if !defined(__APPLE__) || !X86_64
@@ -3201,0 +3203 @@
+#endif
@@ -4756,0 +4759 @@
+#if !defined(__APPLE__) || !X86_64
@@ -5450,0 +5454 @@
+#endif
